spec/user_spec.rb

```diff
@@ -66,18 +171,278 @@
+        end
+
+        it "max not reached when checking against same experiment but another version" do
+          expect(@subject.max_experiments_reached?("link_color:2")).to be_truthy
+        end
+      end
+    end
+
+    context "when multiple experiments with control are allowed" do
+      let(:alternatives) { [ "control", "blue" ] }
+      let(:experiment2) { Split::Experiment.new('link_shape') }
+
+      before do
+        Split.configuration.allow_multiple_experiments = "control"
+        Split::ExperimentCatalog.find_or_create("link_color", alternatives)
+        Split::ExperimentCatalog.find_or_create("link_shape", alternatives)
+      end
+
+      context "user is not in any experiments" do
+        let(:user_keys) { {
+        } }
+
+        it "max not reached for experiment1" do
+          expect(@subject.max_experiments_reached?("link_color")).to be_falsey
+        end
+
+        it "max not reached for experiment2 with a different version" do
+          expect(@subject.max_experiments_reached?("link_shape:2")).to be_falsey
+        end
+      end
+
+      context "user is in control with experiment2" do
+        let(:user_keys) { {
+          'link_shape' => 'control',
+          'link_shape:finished' => true,
+          'link_shape:time_of_assignment' => Time.now.to_s,
+          'link_shape:eligibility' => "ELIGIBLE",
+        } }
+
+        it "max not reached for experiment1" do
+          expect(@subject.max_experiments_reached?("link_color")).to be_falsey
+        end
+
+        it "max not reached for experiment2 with a different version" do
+          expect(@subject.max_experiments_reached?("link_shape:2")).to be_falsey
+        end
+      end
+
+      context "user is in alternative with experiment1, control with experiment2" do
+        let(:user_keys) { {
+          'link_color' => 'blue',
+          'link_color:finished' => true,
+          'link_color:time_of_assignment' => Time.now.to_s,
+          'link_color:eligibility' => "ELIGIBLE",
+          'link_shape' => 'control',
+          'link_shape:finished' => true,
+          'link_shape:time_of_assignment' => Time.now.to_s,
+          'link_shape:eligibility' => "ELIGIBLE",
+        } }
+
+        it "max not reached for experiment1" do
+          expect(@subject.max_experiments_reached?("link_color:1")).to be_falsey
+        end
+
+        it "max reached for experiment2" do
+          expect(@subject.max_experiments_reached?("link_shape:1")).to be_truthy
+        end
+
+        it "max reached for other experiments" do
+          expect(@subject.max_experiments_reached?("link_font")).to be_truthy
+        end
+      end
+    end
+
+    context "when multiple experiments are allowed" do
+      let(:alternatives) { [ "control", "blue" ] }
+      let(:experiment2) { Split::Experiment.new('link_shape') }
+
+      before do
+        Split.configuration.allow_multiple_experiments = true
+        Split.configuration.allow_multiple_experiments = "control"
```

Comment:
Is this correct? 

(It's been a while since i've looked at it and please do correct me if I'm wrong) I believe there's a difference between control and true where when control it allows the user to participate only if the user is in the control group for other experiments (the user can only be in the alternative for one experiment). When true it allows unrestricted participation to alternatives, and when false it only allows 1 experiment at a time.

Reply:
Oops, forgot to take out the assignment to true. It will be overwritten by `control` in Ln356. Yes, the user can join multiple experiments if the user always joined as `control`. But only one alternative can be joined as.